nixos/modules/services/misc/anki-sync-server.nix

```
@@ -123,7 +126,7 @@ in {
         Type = "simple";
         DynamicUser = true;
         StateDirectory = name;
-        ExecStart = "${anki-sync-server-run}/bin/anki-sync-server-run";
+        ExecStart = anki-sync-server-run;
```

```suggestion
        ExecStart = lib.getExe anki-sync-server-run;
```
Unless we have special handling for packages.

---

This is not the package, but this variable: ` anki-sync-server-run = pkgs.writeShellScript "anki-sync-server-run" ''`

I believe the current code is correct?

(that script itself is using getExe in `exec ${lib.getExe cfg.package}` at the end)

---

I would think the result of pkgs.writeShellScript is a regular package, so you do need to specify what to execute exactly.

---

Did you test your approach?

---

I'm not using this as I have another work around, but there's a nixosTests, which I've confirmed just now still passes on this.
The generated content in the vm is as follow:
```
$ systemctl cat anki-sync-server | grep ExecStart
ExecStart=/nix/store/a0jsqs67w3az6r49782wp2icvzph8686-anki-sync-server-run
$ head -n 2 /nix/store/a0jsqs67w3az6r49782wp2icvzph8686-anki-sync-server-run 
#!/nix/store/0irlcqx2n3qm6b1pc9rsd2i8qpvcccaj-bash-5.2p37/bin/bash
# When services.anki-sync-server.users.passwordFile is set,
```

As confusing as the names are, there's writeShellScript which is directly the script (and can be used directly by running the variable) and writeShellScriptBin which is a regular package.


(But thank you for looking at this one year old PR!)